src/main/java/io/lettuce/core/pubsub/PubSubEndpoint.java

```diff
@@ -257,12 +263,21 @@ protected void notifyListeners(PubSubMessage<K, V> message) {
                 case unsubscribe:
                     listener.unsubscribed(message.channel(), message.count());
                     break;
+                case ssubscribe:
+                    shardNotify(listener, (l) -> l.ssubscribed(message.channel(), message.count()));
+                    break;
                 default:
                     throw new UnsupportedOperationException("Operation " + message.type() + " not supported");
             }
         }
     }
 
+    private void shardNotify(RedisPubSubListener<K, V> listener, Consumer<RedisShardedPubSubListener<K, V>> c) {
+        if (listener instanceof RedisShardedPubSubListener) {
```

Comment:
Using `instanceof` to define implementation specific behavior is discouraged as this is not how the object model should be structured. Typically you define the public interface on an interface level and then different implementation override it accordingly. In this case any listener should be able to handle this case otherwise they do not follow the same model and should not be part of the same child-pattern hierarchy.

Reply:
this is one of the things that bugs me. 
getting rid of `instanceof` with use of interface default method.